### PR TITLE
[Actions] Re-instate `brew test-bot` CI checks on PRs (without bottles)

### PR DESCRIPTION
**Pull request by issyl0**
When we fix broken builds, we want to push our fixes to CI and let them run - if it's a long-running or particularly CPU-intensive build, we don't want to always melt our computers rebuilding and rebuilding formulae when GitHub gives us lots of power and time.

### CODE REVIEW

File: .github/workflows/test-formula.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: actions/checkout@master
+      - name: Setup tap
+        run: |
+          rm -rf $(brew --repository ${{github.repository}})
+          ln -s "$GITHUB_WORKSPACE" $(brew --repository ${{github.repository}})
+      - name: brew test-bot
+        run: |
+          brew test-bot \
+            --tap=homebrew/core \
+            --keep-old ${{github.event.client_payload.formula}}
```
Comment (jonchang) [inline .github/workflows/test-formula.yml]:
```suggestion
            --keep-old
```

Comment (issyl0) [reply .github/workflows/test-formula.yml]: Good point - where do we get the formula value from?

Comment (dawidd6) [reply .github/workflows/test-formula.yml]: From git

**Comment by jonchang** [comment]
Hold off on this for now, we can reuse this code for fork-merge workflows.

**Comment by issyl0** [comment]
Superseded by @jonchang's million workflow commits to master.

**Pull request closed without merging**